var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -34,6 +34,9 @@ class Dyninst(CMakePackage):
 
     boost_libs = '+atomic+chrono+date_time+filesystem+system+thread+timer'
 
+    conflicts('^boost@1.70.0:', when='@10.0.0',
```

Comment:
@sknigh some of our tests concretize `dyninst`. This might be old news to you, but `conflict` declarations don't restrict concretization, they just alert the user to errors (at which point they have to update the command line spec e.g. to constrain boost). I think this could also be articulated as a `depends_on` constraint like:

```
depends_on('boost@:1.69.99' + boost_libs, when='@10.0.0')
depends_on('boost@1.61.0:' + boost_libs, when='@:9.99.99') #in this example the existing depends_on for boost also has to be rewritten
```

Or (since this is temporary) you can rewrite the below `depends_on`:

`depends_on('boost@1.61.0:1.69.99' + boost_libs)`

Do either of those options appeal?